src/unix/solarish/mod.rs

```diff
@@ -61,6 +67,53 @@ impl ::Clone for ucred_t {
     }
 }
 
+#[cfg_attr(feature = "extra_traits", derive(Debug))]
+pub enum lgrp_content_t {
```

Comment:
The added enums should probably be annotated `repr()`.  I'm not sure if `repr(C)` would be most appropriate, or an explicitly sized type.

Reply:
`repr(u32)` ?

Reply:
As a general rule, try to avoid using Rust enums to represent C enums. The problem is that Rust is very strict about assuming that an enum will only ever contain the given values and optimizes based on that.

For example in this particular case, Rust wil represent `None` in `Option<lgrp_content_t>` with the value 2 since it knows that this never a valid value for `lgrp_content_t` (which can only be 0 or 1 according to this definition).

Prefer using a typedef for `c_int` and constants instead so that Rust code doesn't break when an enum contains an unexpected value.